juju/juju
# 2.1 containerizer config 1657449

## Description of change

Along with my change to always record all host machine devices, this should enable the original "lxdbr0" functionality that we had in 2.0.

This adds a method on Environments (providers) that we can introspect to see if we think bridging to the host device will be useful. When we would be unable to allocate a Provider level IP address for the containers, we fall back to doing DHCP on the 'lxdbr0' bridge for that container.

## QA steps

```
  $ juju bootstrap aws # or gce or azure
  $ juju switch controller
  $ juju add-machine lxd:0
  $ juju show-machine 0
```

You should end up with a container that has a valid (if local) IP address (most likely 10.0.0.*).

## Bug reference

https://pad.lv/1657449
